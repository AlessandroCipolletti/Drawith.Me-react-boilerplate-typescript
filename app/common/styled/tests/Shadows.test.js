import React from 'react'
import { render } from 'react-testing-library'

// Import de Theme Provider
import { ThemeProvider } from 'styled-components'
import Theme from 'common/Theme'

import {
  SharpShadowCss,
  SharpShadow,
  DiffuseShadowCss,
  DiffuseShadow,
  SoftShadowCss,
  SoftShadow,
  ShorterShadowCss,
  ShorterShadow,
  LongerShadowCss,
  LongerShadow,
} from '../Shadows'

describe('<SharpShadow />', () => {
  const renderComponent = (props = {}) =>
    render(
      <ThemeProvider theme={Theme}>
        <SharpShadow {...props}>
          Content
        </SharpShadow>
      </ThemeProvider>,
    )

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent()

    expect(firstChild).toMatchSnapshot()
  })

  it('Should have public export SharpShadowCss', () => {
    expect(SharpShadowCss).toBeDefined()
  })
})

describe('<DiffuseShadow />', () => {
  const renderComponent = (props = {}) =>
    render(
      <ThemeProvider theme={Theme}>
        <DiffuseShadow {...props}>
          Content
        </DiffuseShadow>
      </ThemeProvider>,
    )

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent()

    expect(firstChild).toMatchSnapshot()
  })

  it('Should have public export DiffuseShadowCss', () => {
    expect(DiffuseShadowCss).toBeDefined()
  })
})

describe('<SoftShadow />', () => {
  const renderComponent = (props = {}) =>
    render(
      <ThemeProvider theme={Theme}>
        <SoftShadow {...props}>
          Content
        </SoftShadow>
      </ThemeProvider>,
    )

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent()

    expect(firstChild).toMatchSnapshot()
  })

  it('Should have public export SoftShadowCss', () => {
    expect(SoftShadowCss).toBeDefined()
  })
})

describe('<ShorterShadow />', () => {
  const renderComponent = (props = {}) =>
    render(
      <ThemeProvider theme={Theme}>
        <ShorterShadow {...props}>
          Content
        </ShorterShadow>
      </ThemeProvider>,
    )

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent()

    expect(firstChild).toMatchSnapshot()
  })

  it('Should have public export ShorterShadowCss', () => {
    expect(ShorterShadowCss).toBeDefined()
  })
})

describe('<LongerShadow />', () => {
  const renderComponent = (props = {}) =>
    render(
      <ThemeProvider theme={Theme}>
        <LongerShadow {...props}>
          Content
        </LongerShadow>
      </ThemeProvider>,
    )

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent()

    expect(firstChild).toMatchSnapshot()
  })

  it('Should have public export LongerShadowCss', () => {
    expect(LongerShadowCss).toBeDefined()
  })
})
