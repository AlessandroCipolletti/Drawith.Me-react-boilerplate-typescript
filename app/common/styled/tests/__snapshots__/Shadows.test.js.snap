// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiffuseShadow /> Should render and match the snapshot 1`] = `
<div
  class="Shadows__DiffuseShadow-sc-1xiu5e5-1 fwpJOJ"
>
  Content
</div>
`;

exports[`<LongerShadow /> Should render and match the snapshot 1`] = `
<div
  class="Shadows__LongerShadow-sc-1xiu5e5-4 daknFo"
>
  Content
</div>
`;

exports[`<SharpShadow /> Should render and match the snapshot 1`] = `
<div
  class="Shadows__SharpShadow-sc-1xiu5e5-0 dbPLDy"
>
  Content
</div>
`;

exports[`<ShorterShadow /> Should render and match the snapshot 1`] = `
<div
  class="Shadows__ShorterShadow-sc-1xiu5e5-3 leWVNE"
>
  Content
</div>
`;

exports[`<SoftShadow /> Should render and match the snapshot 1`] = `
<div
  class="Shadows__SoftShadow-sc-1xiu5e5-2 kjEc"
>
  Content
</div>
`;
